src/http/local_async_normal.hpp

```diff
@@ -254,177 +267,213 @@ struct http_async_connection<Tag, version_major, version_minor>
-                             &this_type::handle_received_data,
-                             this_type::shared_from_this(), status_message,
-                             get_body, callback, placeholders::error,
-                             placeholders::bytes_transferred)),
+              delegate_, request_strand_.wrap([=] (boost::system::error_code const &,
+                                                   std::size_t bytes_transferred) {
+                                                self->handle_received_data(status_message, get_body, callback,
+                                                                           ec, bytes_transferred);
+                                              }),
               bytes_transferred);
-          if (!parsed_ok || indeterminate(parsed_ok)) return;
+          if (!parsed_ok || indeterminate(parsed_ok)) {
+            return;
+          }
         case headers:
           if (ec == boost::asio::error::eof) return;
-          // In the following, remainder is the number of bytes that
-          // remain
-          // in the buffer. We need this in the body processing to make
-          // sure
-          // that the data remaining in the buffer is dealt with before
-          // another call to get more data for the body is scheduled.
-          fusion::tie(parsed_ok, remainder) = this->parse_headers(
+          // In the following, remainder is the number of bytes that remain in
+          // the buffer. We need this in the body processing to make sure that
+          // the data remaining in the buffer is dealt with before another call
+          // to get more data for the body is scheduled.
+          std::tie(parsed_ok, remainder) = this->parse_headers(
               delegate_,
-              request_strand_.wrap(boost::bind(
-                  &this_type::handle_received_data,
-                  this_type::shared_from_this(), headers, get_body, callback,
-                  placeholders::error, placeholders::bytes_transferred)),
+              request_strand_.wrap([=] (boost::system::error_code const &ec,
+                                        std::size_t bytes_transferred) {
+                                     self->handle_received_data(headers, get_body, callback,
+                                                                ec, bytes_transferred);
+                                   }),
               bytes_transferred);
 
-          if (!parsed_ok || indeterminate(parsed_ok)) return;
+          if (!parsed_ok || indeterminate(parsed_ok)) {
+            return;
+          }
 
           if (!get_body) {
-            // We short-circuit here because the user does not
-            // want to get the body (in the case of a HEAD
-            // request).
+            // We short-circuit here because the user does not want to get the
+            // body (in the case of a HEAD request).
             this->body_promise.set_value("");
+            if ( callback )
```

Comment:
Remove extra spacing around callback? Also, this is the first time I've noticed a single condition if without using the braces. If the readability guidelines are "choose what's best for you" around this, can we keep the braces? Single line conditionals have burned me so many times in the past.

Reply:
Ditto here -- this is copied from [boost/network/protocol/http/client/connection/async_normal.hpp](https://github.com/cpp-netlib/cpp-netlib/blob/e80404ec3222b0bd516ea4ad475c66c7222a4350/boost/network/protocol/http/client/connection/async_normal.hpp).